adhocracy4/maps/mixins.py

```diff
@@ -63,13 +62,24 @@ def unpack_geojson(self, data):
 
 class PointFormMixin(GeoJsonPointMixin):
 
-    def get_form_kwargs(self):
-        kwargs = super().get_form_kwargs()
-
+    def __init__(self, *args, **kwargs):
+        # extract geometry from geojson on form submit
+        # FIXME: this should probably be done in a custom form field with clean()
         if "data" in kwargs:
             kwargs["data"] = self.unpack_geojson(kwargs["data"])
+        super().__init__(*args, **kwargs)
 
-        return kwargs
+        # add geojson properties to widget
+        if "data" not in kwargs:
+            properties = self.get_geojson_properties()
+            if properties:
+                instance = kwargs["instance"]
+                props = OrderedDict()
+                for prop, mapping in properties.items():
+                    field = mapping if mapping else prop
```

Comment:
lil confusing here what's happening, what is the mapping or prop expected to be? Maybe add an inline comment